src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java

```diff
@@ -125,6 +126,29 @@ public boolean hasItemStack(@NotNull final Predicate<ItemStack> itemStackSelecti
         return false;
     }
 
+    @Override
+    public boolean hasSimilarStack(@NotNull final ItemStack stack)
+    {
+        final ItemStorage checkItem = new ItemStorage(stack, true);
+        if (content.containsKey(checkItem))
+        {
+            return true;
+        }
+
+        for (final ItemStorage storage : content.keySet())
```

Comment:
I'm wondering if we maybe want to sort by creative tab first and then by tag?

Reply:
I wanted to get rid of the creative tab searching because a lot of that info is client-side only and not "very" reliable.